### ISSUE TITLE
Foundation not playing nice with Rail 6 and Webpacker gem

### ISSUE THREAD

**Issue by salex** [foundation/foundation-sites#11840]
## What should happen?

You should be able to install Foundation in a rails 6 application similar to the approach that the foundation-rails gem does. This is most likely my problem in that I do not fully understanding webpacker and I blame the Rails team for releasing something without at least some guide on how to use webpacker with other components added through gems.

I could of just ignored webpacker and used the asset pipeline, but I'm sure that will disappear in the future, so I thought I'd try.

## What happens instead?

Without a guide, I attempted to put together what little I could find to add Foundation to a barebones rails 6 application using webpacker. After many hours I got my bare requirements (rails6, coffeescript, jQuery, foundation) to semi-work!

My only problem is my last step, adding foundation, disabled all custom coffeescript UJS events from responding. I looked to stackoverflow.com for help with a [stackoverflow question][1] but receive little response. To expand on the stackoverflow question I added the [packdemo][2] application to Github repo so everything is in one place.

  [1]: https://stackoverflow.com/questions/58148550/jquery-events-not-firing-using-webpacker-and-coffeescript.

  [2]: https://github.com/salex/packdemo

## Possible Solution

Again, I probably have the configuration wrong. I did leave turbo links in, which has caused me problems with foundation in the past. But I took it out at one point and still had the non-responsive problem. I've tried to move things around to see if there was a sequence loading problem but nothing helped.

The basic foundation grid works fine (responsive). Don't know if there is a conflict with installing jquery in webpaker and foundation requiring it. Think I did remove the `import "jquery"` and everything except events still work.

## Context

I've been using Foundation for about three years and hope that the reboot is successful. I though I'd try here, hoping for a quick link, something like "Does this _link_ help". If not I try to get on webpackers back!

Thank you for Foundation.

**Comment by DanielRuf** [comment]
Hi @salex,

do you see any errors in the browser console?

Where can we see a livedemo? I have not used RoR for many years.

**Comment by salex** [comment]
There are no errors in the browser console. If I select a button in elements (safari), I can see the click event is loaded.

I put up a development server running code in the repo code. The three buttons don't respond. the refresh button does (rails.ujs). If I remove calls to foundation-sites, all buttons respond.

[Demo](http://stevealex.us:4000)

**Comment by salex** [comment]
Put up another page where I do not have and foundation classes on the page:

[NoFoundationPage](http://stevealex.us/home/nofoundation)

The buttons work! Loading foundation on a page with a class seems to negate other js?

**Comment by DanielRuf** [comment]
Mixing frameworks may have unwanted side-effects.

**Comment by DanielRuf** [comment]
I think it is because of turbolinks.
At least it makes no sense that a click / inspect element on the button in a callout targets the outer container instead of the button.

**Comment by DanielRuf** [comment]
See https://codepen.io/DanielRuf/pen/ExxxdEO in comparison (same markup using Foundation Sites 6.5.3).

**Comment by DanielRuf** [comment]
Or maybe rails itself.

These following code parts are capturing the click:

activestorage.js
```
function didClick(event) {
  var target = event.target;

  if ((target.tagName == "INPUT" || target.tagName == "BUTTON") && target.type == "submit" && target.form) {
    submitButtonsByForm.set(target.form, target);
  }
}
```

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Parameters

> capture: A Boolean indicating that events of this type will be dispatched to the registered listener before being dispatched to any EventTarget beneath it in the DOM tree.

**Comment by salex** [comment]
Thanks for your help. At least I have a few things I can try to narrow it down.

I did update the repo. I'll try to get rid of turbo links and see what that does. Then off to Rails.

I post something if I can figure anything out.

**Comment by jakeNiemiec** [comment]
In file `http://stevealex.us:4000/home/pi/apps/packdemo/app/javascript/packs/scss/_settings.scss` line 881:
```scss
.callout {
  z-index: -1;
}
```

This puts the "callout" divs that contain your buttons behind everything else (clicks never reach buttons). 

Remove this and it should work.

**Comment by DanielRuf** [comment]
> In file `http://stevealex.us:4000/home/pi/apps/packdemo/app/javascript/packs/scss/_settings.scss` line 881:
> 
> ```css-scss
> .callout {
>   z-index: -1;
> }
> ```
> 
> This puts the "callout" divs that contain your buttons behind everything else (clicks never reach buttons).
> 
> Remove this and it should work.

Seems this is the case, right. This code makes not much sense ;-)

This is what I have meant with different frameworks may conflict (especially when class names and code change the same elements).

**Comment by salex** [comment]
The _settings file was copied from the app I'm considering rewriting. If you had a top-bar with a pulldown list, it would go behind a callout if there was one at the top of the content. That was a hack fix the problem that someone suggested.

I just picked a callout for the demo since it stood out.  Guess I should of increased the z-index on the pulldown, but that may have broken something else.

Thank you Jake and Daniel for your help. Maybe I can recover some of the time I've spent spinning the last week or so. At least I did figure how how to get foundation installed in rails 6 with webpacker.
